src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -500,6 +503,49 @@ public TypeInfo(AssemblyInfo assembly, TypeDefinitionHandle typeHandle, TypeDefi
                 FullName = Namespace + "." + Name;
             else
                 FullName = Name;
+
+            DebuggerBrowsableFields = GetDebuggerBrowsableFields();
+
+            Dictionary<string, DebuggerBrowsableState?> GetDebuggerBrowsableFields()
+            {
+                var fields = new Dictionary<string, DebuggerBrowsableState?>();
+                foreach (FieldDefinitionHandle field in type.GetFields())
+                {
+                    try
+                    {
+                        var fieldDefinition = metadataReader.GetFieldDefinition(field);
+                        var fieldName = metadataReader.GetString(fieldDefinition.Name);
+                        foreach (var cattr in fieldDefinition.GetCustomAttributes())
+                        {
+                            try
+                            {
+                                var ctorHandle = metadataReader.GetCustomAttribute(cattr).Constructor;
+                                if (ctorHandle.Kind != HandleKind.MemberReference)
+                                    continue;
+                                var container = metadataReader.GetMemberReference((MemberReferenceHandle)ctorHandle).Parent;
+                                var valueBytes = metadataReader.GetBlobBytes(metadataReader.GetCustomAttribute(cattr).Value);
+                                var attributeName = metadataReader.GetString(metadataReader.GetTypeReference((TypeReferenceHandle)container).Name);
+                                if (attributeName != "DebuggerBrowsableAttribute")
+                                    continue;
+                                var state = (DebuggerBrowsableState)valueBytes[2];
+                                if (!Enum.IsDefined(typeof(DebuggerBrowsableState), state))
+                                    continue;
+                                fields.Add(fieldName, state);
+                                break;
+                            }
+                            catch
+                            {
+                                continue;
+                            }
```

Comment:
This block can be extracted to a local function. IIUC, we'll need the same thing for properties too, in which case the same function can be used to get the browsable members.

Reply:
Will using dynamic be considered as a bad practice here? I would prefer to shorten the code and leave the local fun as it is, only adding a parameter *typeCollection* that will be type.GetFields() or type.GetProperties(), depending on the usage purpose. The only type I can find suitable for both these collections is dynamic. Still, I'm not sure if it's a good idea to type the parameter dynamically